hazelcast/src/main/java/com/hazelcast/internal/crdt/CRDTReplicationMigrationService.java

```diff
@@ -140,11 +141,11 @@ public boolean onShutdown(long timeout, TimeUnit unit) {
                 logger.fine("Skipping replication since all CRDTs are replicated");
                 continue;
             }
-            long start = System.nanoTime();
+            long startNanos = Timer.nanos();
             if (!tryProcessOnOtherMembers(replicationOperation.getOperation(), service.getName(), timeoutNanos)) {
                 logger.warning("Failed replication of CRDTs for " + service.getName() + ". CRDT state may be lost.");
             }
-            timeoutNanos -= (System.nanoTime() - start);
+            timeoutNanos -= Timer.nanosElapsed(startNanos);
```

Comment:
I would pull this timeout logic into a seperate PR. Also I'm not sure if Timer is the appropriate abstraction for it.